libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxHybridSpeechPlayer.kt

```diff
@@ -0,0 +1,100 @@
+package com.mapbox.navigation.ui.voice.api
+
+import android.content.Context
+import androidx.annotation.UiThread
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayer
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayerCallback
+import com.mapbox.navigation.ui.base.model.voice.Announcement
+import com.mapbox.navigation.ui.base.model.voice.SpeechState
+import java.util.LinkedList
+import java.util.Queue
+import java.util.concurrent.ConcurrentLinkedQueue
+
+/**
+ * Hybrid implementation of [SpeechPlayer] combining [MapboxOnboardSpeechPlayer] and
+ * [MapboxOffboardSpeechPlayer] speech players.
+ * @property context Context
+ * @property accessToken String
+ * @property language [Locale] language (ISO 639)
+ */
+@UiThread
+class MapboxHybridSpeechPlayer(
+    private val context: Context,
+    private val accessToken: String,
+    private val language: String
+) : SpeechPlayer {
+
+    private val queue: Queue<SpeechState.ReadyToPlay> = ConcurrentLinkedQueue()
+    private val offboardSpeechPlayer: MapboxOffboardSpeechPlayer =
+        MapboxOffboardSpeechPlayer(context, accessToken, language)
+    private val onboardSpeechPlayer: MapboxOnboardSpeechPlayer =
+        MapboxOnboardSpeechPlayer(context, language)
+    private val clientCallbacks: LinkedList<SpeechPlayerCallback> = LinkedList()
```

Comment:
Why is this a LinkedList? It's this supposed to be in sync with the `Queue<SpeechState.ReadyToPlay>`?

Reply:
We know for sure that announcements are played in order so an ordered list in which the elements are FIFO should be enough. Does that make sense?

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/3977#discussion_r576439038